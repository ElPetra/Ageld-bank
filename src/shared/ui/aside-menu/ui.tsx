import cn from 'classnames';

import type { ReactNode } from 'react';

import './styles.scss';

interface Props {
    visible: boolean;
    children: ReactNode;
}

export const AsideMenu = ({ visible, children }: Props) => {
    const asideClass = cn('aside-menu', {
        visible: visible
    });
    return (
        <>
            {visible && <div className='overlay' />}
            <aside className={asideClass}>
                <div className='aside-menu__content'>{children}</div>
            </aside>
        </>
    );
};
