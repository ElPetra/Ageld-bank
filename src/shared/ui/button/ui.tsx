import { memo } from 'react';

import cn from 'classnames';

import type { ButtonHTMLAttributes, ReactNode } from 'react';

import './styles.scss';

interface Props extends ButtonHTMLAttributes<HTMLButtonElement> {
    type?: 'submit' | 'reset' | 'button';
    disabled?: boolean;
    size?: 'medium' | 'large';
    variant?: 'primary' | 'secondary' | 'link';
    width?: 'auto' | 'max';
    status?: 'active' | undefined;
    children: ReactNode;
}

export const Button = memo(
    ({
        type,
        size = 'medium',
        disabled = false,
        variant = 'primary',
        width = 'auto',
        status,
        children,
        ...props
    }: Props) => {
        const buttonClass = cn(size, width, status, {
            button: variant !== 'link',
            [variant]: !disabled
        });
        return (
            <button
                type={type || 'button'}
                disabled={disabled}
                className={buttonClass}
                {...props}
            >
                {children}
            </button>
        );
    }
);
