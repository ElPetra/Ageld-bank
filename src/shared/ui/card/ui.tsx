import type { ReactNode } from 'react';

import './styles.scss';

interface Props {
    children: ReactNode;
    gap?: 'small' | 'medium' | 'large';
    direction?: 'column' | 'row';
    status?: 'active' | '';
}

export const Card = ({
    children,
    gap = 'medium',
    direction = 'column',
    status
}: Props) => (
    <div className={`card ${gap + '-gap'} ${direction} ${status}`}>
        {children}
    </div>
);
