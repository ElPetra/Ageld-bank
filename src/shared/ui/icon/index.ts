export { Icon, Image } from './ui';
export type { SvgIconNames } from './model';
import MapDotSVG from './assets/icons/map-dot.svg';
export { MapDotSVG };
