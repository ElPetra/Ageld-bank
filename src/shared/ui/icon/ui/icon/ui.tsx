import { ImgHTMLAttributes, memo } from 'react';

import { SvgIconNames } from '../../model/types';

export interface Props extends ImgHTMLAttributes<HTMLImageElement> {
    icon: SvgIconNames;
    width?: number;
    height?: number;
    widthAndHeight?: number;
}

export const Icon = memo(({ icon, width = 24, height = 24, widthAndHeight }: Props) =>
    <img src={'src/shared/ui/icon/assets/icons/' + icon + '.svg'} width={widthAndHeight | width}
         height={widthAndHeight | height} alt='' />
);