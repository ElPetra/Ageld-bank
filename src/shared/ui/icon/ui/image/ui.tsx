import { memo } from 'react';

import type { ImgHTMLAttributes } from 'react';

import type { PngImageNames } from '../../model/types';

export interface Props extends ImgHTMLAttributes<HTMLImageElement> {
    image: PngImageNames;
    widthAndHeight?: number;
}

export const Image = memo(({ image, width, height, widthAndHeight }: Props) => (
    <img
        src={image + '.png'}
        width={widthAndHeight || width || 500}
        height={widthAndHeight || height || 500}
        alt=''
    />
));
