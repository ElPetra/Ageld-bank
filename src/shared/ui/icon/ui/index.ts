export { Icon } from './icon';
export { Image } from './image';
