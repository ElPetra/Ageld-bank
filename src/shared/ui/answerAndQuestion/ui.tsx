import React, { useState } from 'react';

import { Icon } from '../icon';
import './style.scss';

interface Props {
    question: string;
    answer: string;
}

export const AnswerAndQuestion: React.FC<Props> = ({ question, answer }) => {
    const [isOpen, setIsOpen] = useState(false);

    const toggleAnswer = () => {
        setIsOpen(prevState => !prevState);
    };

    return (
        <div className='block'>
            <div className='question'>
                <p className='question__text'>{question}</p>
                <button className='question__button' onClick={toggleAnswer}>
                    <Icon
                        width={24}
                        icon={isOpen ? 'arrow-up' : 'arrow-down'}
                    />
                </button>
            </div>
            {isOpen && (
                <div className='answer'>
                    <p className='answer__text'>{answer}</p>
                </div>
            )}
        </div>
    );
};
