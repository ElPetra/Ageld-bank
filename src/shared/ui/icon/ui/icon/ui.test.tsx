import { render, screen } from '@testing-library/react';

import '@testing-library/jest-dom';
import '@testing-library/jest-dom/jest-globals';

import { Icon } from './ui';

describe('Icon ui', () => {
    test('match snapshot', () => {
        render(<Icon icon='action-add-card' shouldExistTestId />);
        const icon = screen.getByTestId('action-add-card');
        expect(icon).toMatchSnapshot();
    });

    test('Icon renders', () => {
        render(<Icon icon='action-add-card' shouldExistTestId />);
        const icon = screen.getByTestId('action-add-card');
        expect(icon).toBeInTheDocument();
    });
});
