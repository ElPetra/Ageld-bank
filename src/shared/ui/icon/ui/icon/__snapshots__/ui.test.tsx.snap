// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon ui match snapshot 1`] = `
<img
  alt=""
  data-testid="action-add-card"
  height="24"
  width="24"
/>
`;
